packages/events/TopLevelEventTypes.js

```diff
@@ -0,0 +1,11 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+// eslint-disable-next-line no-undef
+export opaque type TopLevelType = number | string;
```

Comment:
I think this should be modeled as a union of all the top level types. See [shared/ReactTypeOfWork.js#L10](https://github.com/facebook/react/blob/master/packages/shared/ReactTypeOfWork.js#L10) for an example. It's more verbose, but that's not a big deal here.